internal/data/testdata/log.go

```diff
@@ -0,0 +1,326 @@
+	rs0lr0 := rs0.Logs().At(0)
+	fillLogOne(rs0lr0)
+	return ld
+}
+
+func generateLogOtlpOneLogNoResource() []*otlplogs.ResourceLogs {
+	return []*otlplogs.ResourceLogs{
+		{
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogOne(),
+			},
+		},
+	}
+}
+
+func GenerateLogDataOneLog() data.Logs {
+	ld := GenerateLogDataOneEmptyLogs()
+	rs0lr0 := ld.ResourceLogs().At(0)
+	rs0lr0.Logs().Resize(1)
+	fillLogOne(rs0lr0.Logs().At(0))
+	return ld
+}
+
+func generateLogOtlpOneLog() []*otlplogs.ResourceLogs {
+	return []*otlplogs.ResourceLogs{
+		{
+			Resource: generateOtlpResource1(),
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogOne(),
+			},
+		},
+	}
+}
+
+func GenerateLogDataOneLogOneNil() data.Logs {
+	return data.LogsFromProto(generateLogOtlpOneLogOneNil())
+}
+
+func generateLogOtlpOneLogOneNil() []*otlplogs.ResourceLogs {
+	return []*otlplogs.ResourceLogs{
+		{
+			Resource: generateOtlpResource1(),
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogOne(),
+				nil,
+			},
+		},
+	}
+}
+
+func GenerateLogDataTwoLogsSameResource() data.Logs {
+	ld := GenerateLogDataOneEmptyLogs()
+	rs0 := ld.ResourceLogs().At(0)
+	rs0.Logs().Resize(2)
+	fillLogOne(rs0.Logs().At(0))
+	fillLogTwo(rs0.Logs().At(1))
+	return ld
+}
+
+// GenerateLogOtlpSameResourceTwologs returns the OTLP representation of the GenerateLogOtlpSameResourceTwologs.
+func GenerateLogOtlpSameResourceTwoLogs() []*otlplogs.ResourceLogs {
+	return []*otlplogs.ResourceLogs{
+		{
+			Resource: generateOtlpResource1(),
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogOne(),
+				generateOtlpLogTwo(),
+			},
+		},
+	}
+}
+
+func GenerateLogDataTwoLogsSameResourceOneDifferent() data.Logs {
+	ld := data.NewLogs()
+	ld.ResourceLogs().Resize(2)
+	rl0 := ld.ResourceLogs().At(0)
+	initResource1(rl0.Resource())
+	rl0.Logs().Resize(2)
+	fillLogOne(rl0.Logs().At(0))
+	fillLogTwo(rl0.Logs().At(1))
+	rl1 := ld.ResourceLogs().At(1)
+	initResource2(rl1.Resource())
+	rl1.Logs().Resize(1)
+	fillLogThree(rl1.Logs().At(0))
+	return ld
+}
+
+func GenerateLogDataManyLogsSameResource(logsCount int) data.Logs {
+	ld := GenerateLogDataOneEmptyLogs()
+	rs0 := ld.ResourceLogs().At(0)
+	rs0.Logs().Resize(logsCount)
+	for i := 0; i < logsCount; i++ {
+		fillLogOne(rs0.Logs().At(i))
+	}
+	return ld
+}
+
+func generateLogOtlpTwoLogsSameResourceOneDifferent() []*otlplogs.ResourceLogs {
+	return []*otlplogs.ResourceLogs{
+		{
+			Resource: generateOtlpResource1(),
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogOne(),
+				generateOtlpLogTwo(),
+			},
+		},
+		{
+			Resource: generateOtlpResource2(),
+			Logs: []*otlplogs.LogRecord{
+				generateOtlpLogThree(),
+			},
+		},
+	}
+}
+
+func fillLogOne(log pdata.LogRecord) {
+	log.SetShortName("logA")
+	log.SetTimestamp(TestLogTimestamp)
+	log.SetDroppedAttributesCount(1)
+	log.SetSeverityNumber(otlplogs.SeverityNumber_INFO)
+	log.SetSeverityText("Info")
+
+	attrs := log.Attributes()
+	attrs.InsertString("app", "server")
+	attrs.InsertString("env", "prod")
```

Comment:
Perhaps use a different attribute type (e.g. int instead of string) for the second attribute for more diverse coverage?